### ISSUE TITLE
GDB Crashes

### ISSUE THREAD

**Issue by matanui159** [Alexpux/MINGW-packages#1862]
I'm not sure if this is the right repository to put this issue in but it's the closest I could find. If I'm wrong, please correct me.

I was originally having problems with GDB crashing in Eclipse, so to test it I used this code:

``` c
#include <stdio.h>

void error(int* ptr) {
    printf("This will cause an error: %i", *ptr);
}

int main(int argc, char* argv[]) {
    error(NULL);
}
```

Compiled it with `gcc -g3 -o test.exe main.c`
Then ran `gdb test.exe` then `run` but this is the result (removed some information for privacy):

```
GNU gdb (GDB) 7.12
Copyright (C) 2016 Free Software Foundation, Inc.
License GPLv3+: GNU GPL version 3 or later <http://gnu.org/licenses/gpl.html>
This is free software: you are free to change and redistribute it.
There is NO WARRANTY, to the extent permitted by law.  Type "show copying"
and "show warranty" for details.
This GDB was configured as "i686-w64-mingw32".
Type "show configuration" for configuration details.
For bug reporting instructions, please see:
<http://www.gnu.org/software/gdb/bugs/>.
Find the GDB manual and other documentation resources online at:
<http://www.gnu.org/software/gdb/documentation/>.
For help, type "help".
Type "apropos word" to search for commands related to "word"...
Reading symbols from test.exe...done.
(gdb) Starting program: C:\...\test.exe 
[New Thread 33072.0x77ac]
[New Thread 33072.0x1630]

Thread 1 received signal SIGSEGV, Segmentation fault.
0x00401639 in error (
This application has requested the Runtime to terminate it in an unusual way.
Please contact the application's support team for more information.
```

I found out through another project that:
 A. its not my program that's crashing and somehow GDB isn't catching it. In my other project, it caught the crash, however, as soon as I called `bt`, it crashed.
 B. it seems to crash trying to load the part in-between the brackets. When it displayed information about my other project it showed `0x00000000 in ?? ()` first, then I called `bt` and it showed the first trace up until the bracket (like above) before crashing.

If you want to see the results of the other project, it is here:

```
GNU gdb (GDB) 7.12
Copyright (C) 2016 Free Software Foundation, Inc.
License GPLv3+: GNU GPL version 3 or later <http://gnu.org/licenses/gpl.html>
This is free software: you are free to change and redistribute it.
There is NO WARRANTY, to the extent permitted by law.  Type "show copying"
and "show warranty" for details.
This GDB was configured as "i686-w64-mingw32".
Type "show configuration" for configuration details.
For bug reporting instructions, please see:
<http://www.gnu.org/software/gdb/bugs/>.
Find the GDB manual and other documentation resources online at:
<http://www.gnu.org/software/gdb/documentation/>.
For help, type "help".
Type "apropos word" to search for commands related to "word"...
Reading symbols from test...done.
(gdb) Starting program: C:\...\test.exe 
[New Thread 32544.0x3fa0]
[New Thread 32544.0x63fc]
[New Thread 32544.0x7584]
[New Thread 32544.0x7d00]

Thread 1 received signal SIGSEGV, Segmentation fault.
0x00000000 in ?? ()
(gdb) #0  0x00000000 in ?? ()
#1  0x00ee192c in getButtonState@8 (
```

I am used Windows 10 64-bit with MinGW w64 i686 fully updated (`pacman -Syuu`)

**Comment by mati865** [comment]
i686 GDB 7.12.0 is broken right now, x86_64 GDB is fine.

**Comment by matanui159** [comment]
I tried using GDB x86_64 (`mingw-w64-x86_64-gdb`) but it has error messages about libraries like `wow64.dll` being `i386:x86_64` and not compatible with target architecture `i386`

Although it still continues on after those messages, it produces weird results which I can only guess are due to those errors.

**Comment by sdhawley** [comment]
I'm seeing the same issue with a simple hello world test program, compiled with the 32 bit gcc. I start "gdb test.exe" and type start and gdb crashes trying to break at main

**Comment by matanui159** [comment]
Not to be a nuisance or anything, but is there any update on this?
I'm finding it very hard to track down bugs without a debugger (since 32-bit GDB doesn't work and I can't seem to get 64-bit GDB to work for a 32-bit program)

**Comment by jtanx** [comment]
Just use [equation's](http://www.equation.com/servlet/equation.cmd?fa=gdb) builds of gdb. I switched to using that after the msys2 builds stopped working (more than a year ago!), and it's been fine.

**Comment by mati865** [comment]
You can try building GDB from git but probably patches need to be updated.
As you can see debugging a debugger is not an easy task so testing trunk version is a first step to go.

**Comment by EmilyBjoerk** [comment]
Also affected by this, could we rollback to 7.11 until this is fixed?

### Here is a workaround for those who previously had GDB 7.11 installed and haven't cleared their pacman cache:

```
# Go to the pacman cache dir
$ cd /var/cache/pacman/pkg

# Find what versions you have from earlier
$ ls | grep gdb-
gdb-7.9-2-x86_64.pkg.tar.xz
mingw-w64-i686-gdb-7.11.1-3-any.pkg.tar.xz
mingw-w64-i686-gdb-7.11.1-4-any.pkg.tar.xz
mingw-w64-i686-gdb-7.12-1-any.pkg.tar.xz
mingw-w64-x86_64-gdb-7.11.1-3-any.pkg.tar.xz
mingw-w64-x86_64-gdb-7.11.1-4-any.pkg.tar.xz
mingw-w64-x86_64-gdb-7.12-1-any.pkg.tar.xz

# Install specific versions (I choose to downgrade 64bit version 
# too because CLion doesn't support it yet)
$ pacman -U mingw-w64-i686-gdb-7.11.1-4-any.pkg.tar.xz
$ pacman -U mingw-w64-x86_64-gdb-7.11.1-4-any.pkg.tar.xz

# Prevent pacman from updating these until problems are resolved
$ nano /etc/pacman.conf

# Add these lines to the file:
IgnorePkg   = mingw-w64-i686-gdb
IgnorePkg   = mingw-w64-x86_64-gdb # If you want to prevent 64 bit from upgrading too

# Verify that it is working:
$ pacman -Syuu
warning: mingw-w64-i686-gdb: ignoring package upgrade (7.11.1-4 => 7.12-1)
warning: mingw-w64-x86_64-gdb: ignoring package upgrade (7.11.1-4 => 7.12-1)

```

**Comment by lazka** [comment]
the old version is still online as well: http://repo.msys2.org/mingw/i686/mingw-w64-i686-gdb-7.11.1-4-any.pkg.tar.xz

**Comment by filnet** [comment]
Just firing gdg and issuing a break command will crash gdg:

```
$ gdb
GNU gdb (GDB) 7.12
Copyright (C) 2016 Free Software Foundation, Inc.
License GPLv3+: GNU GPL version 3 or later <http://gnu.org/licenses/gpl.html>
This is free software: you are free to change and redistribute it.
There is NO WARRANTY, to the extent permitted by law.  Type "show copying"
and "show warranty" for details.
This GDB was configured as "i686-w64-mingw32".
Type "show configuration" for configuration details.
For bug reporting instructions, please see:
<http://www.gnu.org/software/gdb/bugs/>.
Find the GDB manual and other documentation resources online at:
<http://www.gnu.org/software/gdb/documentation/>.
For help, type "help".
Type "apropos word" to search for commands related to "word".
(gdb) break

This application has requested the Runtime to terminate it in an unusual way.
Please contact the application's support team for more information.
```

**Comment by aquiles2k** [comment]
Im not a pacman gutu so could anyone please just post some commandlines how to downgrade the gdb to the linked 7.11 package. gdb is pretty essential for me when developing ...

**Comment by lazka** [comment]
```
wget http://repo.msys2.org/mingw/i686/mingw-w64-i686-gdb-7.11.1-4-any.pkg.tar.xz
pacman -U mingw-w64-i686-gdb-7.11.1-4-any.pkg.tar.xz
```

**Comment by aquiles2k** [comment]
Thanks, works like a charm!

**Comment by lazka** [comment]
Likely fixed (see #1862)
